presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestDetermineJoinDistributionType.java

```diff
@@ -238,6 +240,40 @@ public void testFlipAndReplicateWhenOneTableMuchSmaller()
                         values(ImmutableMap.of("A1", 0))));
     }
 
+    @Test
+    public void testFlipAndReplicateWhenOneTableMuchSmallerAndJoinCardinalityUnknown()
```

Comment:
Add similar test for `flip and repartition`, e.g: basic case when one table is smaller, but only like 2x and not 100x?

Reply:
The main reason for additional tests is to verify that the cost based decision can be made even when output size of a join cannot be estimated. Do you think it is worth adding yet another test for that purpose?